yt/utilities/lib/interpolators.pyx

```diff
@@ -163,3 +163,115 @@ def ghost_zone_interpolate(int rf,
                 opos[2] += ods[2]
             opos[1] += ods[1]
         opos[0] += ods[0]
+
+@cython.cdivision(True)
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def replace_nonperiodic_with_extrap(np.ndarray[np.float64_t, ndim=3] field,
+                                    np.ndarray[np.int_t,     ndim=1] ncell_lo,
+                                    np.ndarray[np.int_t,     ndim=1] ncell_hi):
+    """
+    The purpose of this function is to extrapolate data from the interior of
+    the buffer region to replace the given data in the buffer region using a
+    first-order approximation.
+
+    Parameters
+    ----------
+    field : 3D array
+        Three-dimensional data of the flowfield
+    ncell_lo : 1D array of length 3
+        Number of buffer cells to replace on the low side along each axis
+    ncell_hi : 1D array of length 3
+        Number of buffer cells to replace on the high side along each axis
+
+    This is best illustrated through a one-dimensional example. Consider a
+    linear function: f(x) = x , and the `field` variable was filled using
+    periodicity, so we have the following data:
```

Comment:
better be precise on what you mean by "filled" here (I'm assuming that you mean "values on each end of an array", or something of the essence)